File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamfriends/callback/PersonaChangeCallback.java
package in.dragonbra.javasteam.steam.handlers.steamfriends.callback;

import in.dragonbra.javasteam.enums.EPersonaState;
import in.dragonbra.javasteam.enums.EResult;
import in.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserverFriends.CMsgPersonaChangeResponse;
import in.dragonbra.javasteam.steam.handlers.steamfriends.SteamFriends;
import in.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg;
import in.dragonbra.javasteam.types.JobID;

/**
 * This callback is fired in response to setting this client's persona name or state
 * with {@link SteamFriends#setPersonaName(String)} or {@link SteamFriends#setPersonaState(EPersonaState)}.
 */
public class PersonaChangeCallback extends CallbackMsg {

    private final EResult result;

    private final String name;

    public PersonaChangeCallback(JobID jobID, CMsgPersonaChangeResponse.Builder msg) {
        setJobID(jobID);

        result = EResult.from(msg.getResult());
        name = msg.getPlayerName();
    }

    /**
     * @return the result of changing this client's persona information.
     */
    public EResult getResult() {
        return result;
    }

    /**
     * @return the name of this client according to Steam.
     */
    public String getName() {
        return name;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamfriends/callback/PersonaChangeCallback.kt
package `in`.dragonbra.javasteam.steam.handlers.steamfriends.callback

import `in`.dragonbra.javasteam.enums.EResult
import `in`.dragonbra.javasteam.protobufs.steamclient.SteammessagesClientserverFriends.CMsgPersonaChangeResponse
import `in`.dragonbra.javasteam.steam.handlers.steamfriends.SteamFriends
import `in`.dragonbra.javasteam.steam.steamclient.callbackmgr.CallbackMsg
import `in`.dragonbra.javasteam.types.JobID

/**
 * This callback is fired in response to setting this client's persona name or state
 * with [SteamFriends.setPersonaName] or [SteamFriends.setPersonaState].
 */
class PersonaChangeCallback(jobID: JobID, msg: CMsgPersonaChangeResponse.Builder) : CallbackMsg() {

    /**
     * Gets the result of changing this client's persona information.
     */
    val result: EResult

    /**
     * Gets the name of this client according to Steam.
     */
    val name: String

    init {
        this.jobID = jobID

        result = EResult.from(msg.result)
        name = msg.playerName
    }
}
